restapi/src/main/java/io/stargate/web/docsapi/service/query/FilterPath.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package io.stargate.web.docsapi.service.query;
+
+import java.util.Collections;
+import java.util.List;
+import org.immutables.value.Value;
+
+/** Contains path information for a filter operation. */
+@Value.Immutable
+public interface FilterPath {
+
+  /** @return The complete path to the field, including the field name as the last element. */
+  @Value.Parameter
+  List<String> getFullPath();
+
+  /** @return The name of the field. Effectively, the last element of {@link #getPath()}. */
+  default String getField() {
+    List<String> path = getFullPath();
+    if (null == path || path.isEmpty()) {
+      return null;
```

Comment:
Is this not an error to have a `FilterPath` without any path elements? Even a field at the root level should have one path element `==` the field's name, right?

Reply:
Good catch, adding a check.. And changing the impl everywhere!